File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hmap;
        for(const auto& s : strs) {
            string key = s;
            sort(key.begin(), key.end());
            hmap[key].push_back(s);
        }
        vector<vector<string>> res;
        for(const auto& pair : hmap) {
            res.push_back(pair.second);
        }
        return res;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashMap = defaultdict(list)

        for item in strs:
            key = "".join(sorted(item))
            hashMap[key].append(item)
            
        res = []
        for key, value in hashMap.items():
            res.append(value)

        return res